src/core/model.jl

```diff
@@ -101,32 +128,43 @@ end
 """
     agent_validator(agent, space)
 Validate the user supplied agent (subtype of `AbstractAgent`).
-Checks for mutability and existance and correct types for fields depending on `SpaceType`.
+Checks for mutability and existence and correct types for fields depending on `SpaceType`.
 """
 function agent_validator(::Type{A}, space::S) where {A<:AbstractAgent, S<:SpaceType}
     # Check A for required properties & fields
-    isconcretetype(A) || throw(ArgumentError("Agent struct must be a concrete type"))
-    isbitstype(A) && throw(ArgumentError("Agent struct must be mutable"))
-    (any(isequal(:id), fieldnames(A)) && fieldnames(A)[1] == :id) || throw(ArgumentError("First field of Agent struct must be `id`"))
+    if VERSION < v"1.1"
+        isconcrete = true
+        for f in 1:fieldcount(A)
+            if !isconcretetype(fieldtype(A, f))
+                isconcrete = false
+                break
+            end
+        end
+    else
+        iscontrete = all(isconcretetype, fieldtypes(A))
+    end
+    isconcrete || throw(ArgumentError("Agent struct must be a concrete type. If your agent is parametrically typed, this probably happened because you gave `Agent` instead of `Agent{T}` to this function. You can also create and instance of your agent and pass it to this function."))
+    isbitstype(A) && throw(ArgumentError("Agent struct must be mutable. Try adding the `mutable` keyword infront of `struct` in your agent definition."))
+    (any(isequal(:id), fieldnames(A)) && fieldnames(A)[1] == :id) || throw(ArgumentError("First field of Agent struct must be `id` (it should be of type `Integer`)."))
     if space != nothing
-        (any(isequal(:pos), fieldnames(A)) && fieldnames(A)[2] == :pos) || throw(ArgumentError("Second field of Agent struct be `pos` when using a space"))
+        (any(isequal(:pos), fieldnames(A)) && fieldnames(A)[2] == :pos) || throw(ArgumentError("Second field of Agent struct be `pos` when using a space (it should be of type `NTuple{Integer}`)."))
```

Comment:
This is incorrect. If `GraphSpace` is used, type `Int` is used for the position. You don't need the added sentence `(it shoud...)` here, since you already take care of it below.

Reply:
Yep, an oversight.